src/main/java/net/sf/jabref/logic/util/DOI.java

```diff
@@ -107,6 +107,23 @@ public DOI(String doi) {
     }
 
     /**
+     * Creates an Optional<DOI> from various schemes including URL, URN, and plain DOIs.
+     *
+     * Useful for suppressing the <c>IllegalArgumentException</c> of the Constructor
+     * and checking for Optional.isPresent() instead.
+     *
+     * @param doi the DOI string
+     * @return an Optional containing the DOI or an empty Optional
+     */
+    public static Optional<DOI> build(Optional<String> doi) {
```

Comment:
Better would be if you have an optional string to use `optionalString.flatMap(string -> DOI.build(string))` as this would then return an Optional<DOI>, and the code would only fire if there is a string.